src/components/PhoneInput/countriesDropdown/index.js

```diff
@@ -0,0 +1,136 @@
+import React, { useRef, memo } from 'react';
+import PropTypes from 'prop-types';
+import {
+    useFilterCountries,
+    useKeyboardNavigation,
+    useScrollControls,
+    useItemsRef,
+    useHandleCountryChange,
+} from './hooks';
+import {
+    StyledScrollable,
+    StyledUl,
+    StyledDropdown,
+    StyledSearchContainer,
+    StyledSearch,
+    StyledScrollControls,
+    StyledMenuArrowButton,
+} from './styled';
+import CountriesList from './countriesList';
+import RenderIf from '../../RenderIf';
+import StyledSearchIcon from '../../Lookup/options/styled/searchIcon';
+import StyledOptionsContainer from '../../Lookup/options/styled//optionsContainer';
+import StyledEmptyMessage from '../../Lookup/options/styled/emptyMessage';
+
+const CountriesDropdown = memo(props => {
+    const {
+        country,
+        countries,
+        isOpen,
+        searchRef,
+        setFocusIndex,
+        onCountryChange,
+        handleFocus,
+        handleBlur,
+    } = props;
+
+    const scrollableRef = useRef();
+    const [query, countriesFiltered, setQuery] = useFilterCountries(countries, country);
+    const itemsRef = useItemsRef(countriesFiltered.length);
+    const handleCountryChange = useHandleCountryChange(scrollableRef, onCountryChange, setQuery);
+    const handleActiveChange = useKeyboardNavigation(
+        country,
+        countriesFiltered,
+        searchRef,
+        scrollableRef,
+        itemsRef,
+        handleCountryChange,
+        setFocusIndex,
+    );
```

Comment:
for now leave this, but we should avoid pass to much arguments to a function, instead, when they are more than two we pass an object (only one argument with named parameters)